crawler/crawlutils.py

```diff
@@ -458,30 +459,30 @@ def snapshot(
     signal.signal(signal.SIGHUP, signal_handler_exit)
 
     if crawlmode == Modes.OUTCONTAINER:
+	#Use custom filter logic to distribute already running containers
         containers = get_filtered_list_of_containers(options, namespace,
                                                      runtime_env)
 
+    eventQ = options['partition_strategy']['args']['eventQ']
+    delList = options['partition_strategy']['args']['delList']
+
+    if frequency < 0:
+	wait_period = 5
+    else:
+	wait_period = 5
+	
     # This is the main loop of the system, taking a snapshot and sleeping at
     # every iteration.
 
     while True:
-
-        snapshot_time = int(time.time())
-
         if crawlmode == Modes.OUTCONTAINER:
-
-            curr_containers = get_filtered_list_of_containers(
```

Comment:
thats true, for event design to work across non-docker containers, we would need event subscription capability for non-docker providers also. 

i will see if we can add this as a complimentary plugin for event subscription rather than replacement for current polling mechanism